Dockerfile

```
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM registry.access.redhat.com/ubi8/go-toolset:1.20 as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:1.21 as builder
```

@Msarawan your proposal to align all go versions to 1.21.11 that is the latest registry.redhat.io/rhel8/go-toolset makes sense to me because the go-toolset container is used to build the official operator container image. We should use the same versions also in the workflows and doc.
```suggestion
FROM registry.access.redhat.com/ubi8/go-toolset:1.21.11 as builder
```

---

@brusdev @gaohoward updated the go version as per the latest registry.redhat.io/rhel8/go-toolset.